bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/editor/SummernoteEditor.java

```diff
@@ -0,0 +1,160 @@
+package de.agilecoders.wicket.extensions.markup.html.bootstrap.editor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.wicket.ajax.AbstractDefaultAjaxBehavior;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.attributes.AjaxRequestAttributes;
+import org.apache.wicket.markup.head.CssHeaderItem;
+import org.apache.wicket.markup.head.IHeaderResponse;
+import org.apache.wicket.markup.head.JavaScriptHeaderItem;
+import org.apache.wicket.markup.head.OnDomReadyHeaderItem;
+import org.apache.wicket.markup.html.form.FormComponent;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.protocol.http.servlet.MultipartServletWebRequest;
+import org.apache.wicket.protocol.http.servlet.ServletWebRequest;
+import org.apache.wicket.request.http.WebResponse;
+import org.apache.wicket.util.crypt.Base64;
+import org.apache.wicket.util.io.IOUtils;
+import org.apache.wicket.util.lang.Args;
+import org.apache.wicket.util.lang.Bytes;
+import org.apache.wicket.util.template.PackageTextTemplate;
+import org.apache.wicket.util.upload.FileItem;
+import org.apache.wicket.util.upload.FileUploadException;
+
+import com.google.common.io.Files;
+
+/**
+ * 
+ * Please add
+ * 
+ * mountResource(SummernoteStoredImageResourceReference.SUMMERNOTE_MOUNT_PATH,
+ * new SummernoteStoredImageResourceReference(new File("/Users/MyUser/"),"subpath"));
+ * 
+ * and then use config
+ * SummernoteConfig config = new SummernoteConfig();
+ * config.setSubFolderName(subpath");
```

Comment:
The documentation is not finished in general - if the code is ok we should finish it. I only made some short hints, because a lot of changes were done and the doc changes a lot, too